projects/igniteui-angular/src/lib/icon/icon.component.ts

```diff
@@ -32,63 +32,190 @@ export class IgxIconComponent implements OnInit {
     @ViewChild('explicitLigature', { read: TemplateRef })
     private explicitLigature: TemplateRef<HTMLElement>;
 
+    /**
+     *  This allows you to disable `igx-icon` class. By default it's applied.
+     *```html
+     *@ViewChild("MyIcon")
+     *public icon: IgxIconComponent;
+     *ngAfterViewInit() {
+     *    icon.cssClass = false;
+     * }
+     * ```
+     */
     @HostBinding('class.igx-icon')
     public cssClass = 'igx-icon';
 
+    /**
+     *  This allows you to disable the `aria-hidden` attribute. By default it's applied.
+     *```html
+     *@ViewChild("MyIcon")
+     *public icon: IgxIconComponent;
+     *ngAfterViewInit() {
+     *    icon.ariaHidden = false;
```

Comment:
@ViewChild("MyIcon") public icon: IgxIconComponent;

    constructor(private cdRef:ChangeDetectorRef) {}

    ngAfterViewInit() {
        this.icon.ariaHidden = false;
        this.cdRef.detectChanges();
    }

Aria hidden would require invoking `detectChages()` or it would throw an error. And it should be **this.icon**